workflow/metrics/counter_template.go

```diff
@@ -0,0 +1,33 @@
+package metrics
+
+import (
+	"context"
+)
+
+const (
+	nameWFTemplateTriggered = `workflowtemplate_triggered_total`
+)
+
+func addWorkflowTemplateCounter(_ context.Context, m *Metrics) error {
+	return m.createInstrument(int64Counter,
+		nameWFTemplateTriggered,
+		"Total number of workflow templates triggered by workflowTemplateRef",
+		"{workflow_template}",
+		withAsBuiltIn(),
+	)
+}
+
+func templateLabels(name, namespace string, cluster bool) instAttribs {
+	return instAttribs{
+		{name: labelTemplateName, value: name},
+		{name: labelTemplateNamespace, value: namespace},
+		{name: labelTemplateCluster, value: cluster},
+	}
+}
+
+func (m *Metrics) CountWorkflowTemplate(ctx context.Context, phase, name, namespace string, cluster bool) {
```

Comment:
The argument name `phase` is somewhat misleading here in my opinion. 
A comment and an enum for the `phase` argument would make sense here.

Because this is a different type effectively, one that inherits all possible values of the `Phase` field along with an additional `New` field`

Reply:
I've done a new type for this. Still called it `phase` in here, I hope that makes sense.